src/main/java/strategy/impl/BalanceHandler.java

```diff
@@ -0,0 +1,16 @@
+package strategy.impl;
+
+import java.util.Map;
+import strategy.OperationHandler;
+
+public class BalanceHandler implements OperationHandler {
+
+    @Override
+    public void doTransaction(Map<String, Integer> reportList, String fruit, int value) {

```

Comment:
```suggestion
    public void doTransaction(String fruit, int value) {
```
Call the storage map directly, it's a static map  
Apply to all handlers